goversion/goversion_test.go

```diff
@@ -127,3 +128,76 @@ func TestGoVersion_UpstreamFormatGitTag(t *testing.T) {
 		})
 	}
 }
+
+func TestGoVersions_Sort(t *testing.T) {
+	tests := []struct {
+		name     string
+		versions GoVersions
+		expected GoVersions
+	}{
+		{
+			name: "basic version sorting",
+			versions: GoVersions{
+				New("1.2.3"),
+				New("1.2.1"),
+				New("1.3.0"),
+				New("1.2.3-2"),
+				New("1.2.3-1"),
+			},
+			expected: GoVersions{
+				New("1.3.0"),
+				New("1.2.3-2"),
+				New("1.2.3"),
+				New("1.2.3-1"),
+				New("1.2.1"),
+			},
+		},
+		{
+			name: "version with prerelease and note",
+			versions: GoVersions{
+				New("1.2.3-beta"),
+				New("1.2.3-rc1"),
+				New("1.2.3-1-fips"),
+				New("1.2.3"),
+				New("1.2.3-2"),
+			},
+			expected: GoVersions{
+				New("1.2.3-2"),
+				New("1.2.3"),
+				New("1.2.3-beta"),
+				New("1.2.3-1-fips"),
+				New("1.2.3-rc1"),
+			},
+		},
+		{
+			name: "sorting with major and minor versions",
+			versions: GoVersions{
+				New("2.0.0"),
+				New("1.10.0"),
+				New("1.2.3"),
+				New("1.2.0"),
+				New("1.3.0"),
+			},
+			expected: GoVersions{
+				New("2.0.0"),
+				New("1.10.0"),
+				New("1.3.0"),
+				New("1.2.3"),
+				New("1.2.0"),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Sort the versions
+			sort.Sort(tt.versions)
+			for i, v := range tt.versions {
+				if *v != *tt.expected[i] {
+					t.Errorf(tt.name)
```

Comment:
As far as I know the subtest name is always in the log just above a test failure's output, so this isn't necessary.

```suggestion
```